### PR TITLE
ci: cleanup CI config and update setup

### PR DESCRIPTION
**Pull request by xSAVIKx**
In this PR I have updated our CI setup and ensured the latest stable action versions are used.

Also, we're now testing the package against all the latest OS versions to be on the safe side.

### CODE REVIEW

**Comment by xSAVIKx** [comment]
@Klaudioz PTAL.

File: .github/workflows/pypi-release.yml [hunk 1]
```diff
@@ -9,17 +9,18 @@ jobs:
   build-and-publish:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v3
```
Comment (Klaudioz) [inline .github/workflows/pypi-release.yml:14]: Why is this one different from the other changes? (v4)

Comment (xSAVIKx) [reply .github/workflows/pypi-release.yml:14]: A silly mistake. I'll update to v4.

**Review by Klaudioz — APPROVED** [review]
LGTM

**Pull request merged**